### DISCUSSION TITLE
Hash images/files to reduce double files

### DISCUSSION THREAD

**Discussion by Rar9** [payloadcms/payload#1263] [Feature Requests & Ideas]
Is it possible to allow ONE image/file to be uploaded, so NO double image/files via can be uploaded?

I believe this could be done by some hash hook on images/files to reduce storage space?

If so how?

I also believe unsave/discarded uploads  are never removed :-(

**Comment by DanRibbens** [comment]
You could do this fairly simple by adding a field to your upload collection for the hash with:

```js
// field in the collection
{
  name: 'hash',
  type: 'text',
  unique: true,
  hooks: {
  beforeChange: [
  ({siblingdata}) => {
    return md5(siblingData.file)
  }
],
},
}
```

Something like this should work. I'm replying on mobile and haven't had a chance to test this. Let me know how it goes.

**Reply by Rar9** [reply]
Hmm... I added hook to media.ts so the hash is showing, but it is not created when the image is actually added. 

Should the hash not be created on the actual upload image /file field in the content?

![Screenshot_20221016-090000_Firefox.jpg](https://user-images.githubusercontent.com/2239608/196022868-d838f3b8-eed4-456b-a852-74febeb61b80.jpg)

I also require a user thumbnail, so the default the alt/title fields are not required for this specific image upload, and the thumbnail needs a specific transformation.... do I need a separate media_user.ts for this?

**Reply by reiv** [reply]
The `md5` function in @DanRibbens' comment was probably just meant as an example, you'll still have to write it yourself; For example:

```ts
import crypto from 'node:crypto';

export default function md5(buf: Buffer): string {
  const hash = crypto.createHash('md5');
  hash.update(buf);
  const digest = hash.digest();

  return digest.toString('hex');
}
```

Remember to [alias](https://payloadcms.com/docs/admin/webpack#aliasing-server-only-modules) this module in your webpack config to exclude it from the admin bundle, since it won't run in the browser.

Also, if you want `hash` to be a required field, you'll want to populate it in `beforeValidate` instead of `beforeChange`.

**Reply by Rar9** [reply]
So how / where?
